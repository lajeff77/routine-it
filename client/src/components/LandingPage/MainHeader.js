import { AppBar, Button, Toolbar, Typography } from '@mui/material';
import { Link } from 'react-router-dom';
import React from 'react';

const MainHeader = () => {

    return(
        <AppBar position="static" sx={{padding: "1%"}}>
            <Toolbar>
                <Typography variant='h3' component="div" sx={{ flexGrow: 1 }}>Routine It</Typography>            
                <Button component={Link} to="/" variant="contained" disableElevation sx={{margin: "10px"}}>Home</Button>
                <Button component={Link} to="/signup" variant="contained" disableElevation sx={{margin: "10px"}}>Sign Up</Button>
                <Button component={Link} to="/signin" variant="contained" disableElevation sx={{margin: "10px"}}>Sign In</Button>
            </Toolbar>
        </AppBar>
    )
}

export default MainHeader;