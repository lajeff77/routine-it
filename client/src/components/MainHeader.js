import { AppBar, Button, Toolbar, Typography } from '@mui/material';
import React from 'react';
import {Link} from 'react-router-dom';

const MainHeader = () => {

    return(
        <AppBar position="static" sx={{padding: "1%"}}>
            <Toolbar>
                <Typography variant='h3' component="div" sx={{ flexGrow: 1 }}>Routine It</Typography>            
                <Button href="/" variant="contained" disableElevation sx={{margin: "10px"}}>Home</Button>
                <Button href="/signup" variant="contained" disableElevation sx={{margin: "10px"}}>Sign Up</Button>
                <Button href="/signin" variant="contained" disableElevation sx={{margin: "10px"}}>Sign In</Button>
            </Toolbar>
        </AppBar>
    )
}

export default MainHeader;