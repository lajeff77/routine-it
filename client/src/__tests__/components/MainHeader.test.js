import {render } from '@testing-library/react'
import React from "react";
import { MemoryRouter} from "react-router-dom";
import MainHeader from '../../components/MainHeader';

describe(MainHeader.name, ()=>{
  test('MainHeader matches snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <MainHeader />
      </MemoryRouter>
    );
    
    expect(container).toMatchSnapshot();
  });

  test('The links in MainHeader are accessible', () => {
    const { getByText } = render(
    <MemoryRouter>
      <MainHeader />
    </MemoryRouter>);
    const homeLink = getByText('Home');
    const signUpLink = getByText('Sign Up');
    const signInLink = getByText('Sign In');
  
    expect(homeLink.getAttribute('href')).toBe('/');
    expect(signUpLink.getAttribute('href')).toBe('/signup');
    expect(signInLink.getAttribute('href')).toBe('/signin');
  });
});
