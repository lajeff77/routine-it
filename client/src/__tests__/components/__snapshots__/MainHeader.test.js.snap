// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader MainHeader matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      id="main-header"
    >
      <a
        class="header-logo"
        href="/"
      >
        <h1>
          Routine It
        </h1>
      </a>
      <a
        class="header-link"
        href="/"
      >
        <p>
          Home
        </p>
      </a>
      <a
        class="header-link"
        href="/signup"
      >
        <p>
          Sign Up
        </p>
      </a>
      <a
        class="header-link"
        href="/signin"
      >
        <p>
          Sign In
        </p>
      </a>
    </div>
    <div
      class="hero-section"
    >
      <h1>
        Everyone should be able to keep a routine!
      </h1>
      <a
        href="/signup"
      >
        <button>
          Get Started!
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
